src/hotspot/share/runtime/thread.cpp

```diff
@@ -2717,6 +2717,14 @@ jint Threads::check_for_restore(JavaVMInitArgs* args) {
   return JNI_OK;
 }
 
+void Threads::close_extra_descriptors(JavaVMInitArgs *args) {
+  if (Arguments::is_option_set(args, "-XX:CRaCCheckpointTo")) {
```

Comment:
For Checkpoint, we should rely as much as possible on existing argument parsing. Restore was a bit special and without a good soluton and the need for direct access for the arguments, we've injected some CRaC and platform specific code here. But it's better to avoid this as a general practice.